### ISSUE TITLE
Compatibility Issue with IsPrerendering Property Removal in Oqtane v5

### ISSUE THREAD

**Issue by tvatavuk** [oqtane/oqtane.framework#3473]
We encountered an issue related to the `IsPrerendering` property, which seems to have been removed from the `SiteState` class in Oqtane version 5.0.0 (current dev branch). This change can be observed in the commit at [fix #3438 - remove IsPrerendering property as it is not used, and remove reference to HttpContext](https://github.com/oqtane/oqtane.framework/commit/b869308bf41664d605740acbc11742853d469e14).

The absence of the `IsPrerendering` property is causing compatibility issues with the latest 2sxc version 16.07.00 LTS when used in conjunction with Oqtane 5. Normally, significant API changes like this are expected in major version updates.

Our goal is to release 2sxc version 16.08.00, which would be compatible with both the latest Oqtane 4 (specifically 4.0.6) and the upcoming Oqtane 5 versions (until anticipated larger changes in Oqtane 5.1). We are exploring various strategies to maintain this compatibility, including the use of reflection or possibly reimplementing the missing functionality within 2sxc.

We would greatly appreciate any suggestions or guidance on how to best approach this situation. Understanding your perspective on this change https://github.com/oqtane/oqtane.framework/issues/3438 and any recommended strategies for maintaining compatibility across these versions would be extremely helpful.

Thank you for your attention.

**Comment by sbwalker** [comment]
@tvatavuk the IsPrerendering property was relying on HttpContext... and the guidance from Microsoft is that HttpContext should not be used in Blazor applications. The HttpContext reference was causing an exception which was preventing Oqtane from loading in some environments - see  #3438.  Since the IsPrerendering property was not actually used anywhere within the framework, it was deemed safe to remove (it was originally added in 4.x to try and troubleshoot some issues with injecting JavaScript). An alternative solution would be to retain the IsPrerendering property and include a nullable validation check. Essentially this would prevent the exception from being thrown - however it is important to note that the IsPrerendering property value will not be reliable in all environments (ie. it may return False when the application is pre-rendering).

**Comment by sbwalker** [comment]
In .NET 8 Microsoft provided a way to get access to HttpContext... which apparently is better than using IHttpContextAccesor. I will explore today.

**Comment by sbwalker** [comment]
@tvatavuk Based on Microsoft's guidance that IHttpContextAccessor should never be used in razor components, I implemented the new CascadingParameter method in .NET 8 to get the HttpContext. It has a different behavior than IHttpContextAccessor which I did not expect. During pre-rendering the HttpContext is populated... and when it is rendering it is null. I suppose this makes sense as it means when the code is running on the server it has an HttpContext... but once it transitions to interactive rendering it no longer has an HttpContext.

```
    [CascadingParameter]
    HttpContext HttpContext { get; set; }

    SiteState.IsPrerendering = (HttpContext != null) ? true : false;
```

I have read previously that trying to add conditional behavior in your Blazor code related to Pre-rendering is an anti-pattern which should be avoided.  However since I do not want to break the API, I will add this property back for 5.0.
